### PR TITLE
Some refactoring and addition of donor updates

### PR DESCRIPTION
**Pull request by kdawgsf**

### CODE REVIEW

**Comment by kdawgsf** [comment]
@arvindraghavan I think we can roll with this if you're ok with the changes. I didn't do phone number updates, but we can leave that as an exercise for next year.

**Comment by arvindraghavan** [comment]
@kdawgsf, looked through the code, it looks great. I'm pretty much ready to merge, I just want to double check one thing, which I'll do at lunch time. Did you do a mock upload and find everything to be as expected?

**Comment by kdawgsf** [comment]
Yes, it was as much as I could tell from a dry run

File: update_existing_families.py [hunk 1]
```diff
@@ -130,17 +164,20 @@ def save_as_csv_file(filename, header_fields, data):
             studentrecord['SCHOOL'] = 'ALUM'
         elif dp_record['SCHOOL'] != 'ALUM':
             studentrecord['SCHOOL'] = 'NOBSD'
-        if dp_record != studentrecord:
-            dp_import_studentrecords.append(studentrecord)
+        if dp_record_minimal != studentrecord:
```
Comment (arvindraghavan) [inline update_existing_families.py:171]: @kdawgsf, do you mind adding the following in comments in the code: 
`#The two records will be the same if:`
`# (a) we run this mid-year, OR`
`# (b) a student is already NOBSD`
`#The same process works whether we're running a summer update or a mid-year update.`

**Comment by arvindraghavan** [comment]
@kdawgsf, I'm all set with this PR, except for that one comment I'd like to see in the code. I keep forgetting the logic even after a couple of days, so it will be worse in a couple of months when we get the next dump from BSD.

**Comment by kdawgsf** [comment]
@arvindraghavan I added your comment with another commit

**Pull request merged**